hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/security/S3GetSecretRequest.java

```diff
@@ -73,10 +73,14 @@ public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
     String kerberosID = s3GetSecretRequest.getKerberosID();
 
     UserGroupInformation user = ProtobufRpcEngine.Server.getRemoteUser();
+    // Permission check. Users need to be themselves or have admin privilege
     if (!user.getUserName().equals(kerberosID)) {
-      throw new OMException("User mismatch. Requested user name is " +
-          "mismatched " + kerberosID +", with current user " +
-          user.getUserName(), OMException.ResultCodes.USER_MISMATCH);
+      if (!ozoneManager.isAdmin(kerberosID)) {
+        throw new OMException("Requested user name '" + kerberosID +
+            "' doesn't match current user '" + user.getUserName() +
+            "', nor does current user has administrator privilege.",
+            OMException.ResultCodes.USER_MISMATCH);
+      }
```

Comment:
Hey @vivekratnavel thanks for the comment.  But the added `else` wouldn't work because it is triggered when an admin is revoking another user's secret.

Reply:
I have replaced the nested if with a single one to make the logic cleaner, hopefully.

Reply:
@smengcl Thanks for the clarification!